src/librustdoc/clean/mod.rs

```diff
@@ -1273,6 +1271,24 @@ pub(crate) fn clean_middle_assoc_item<'tcx>(
                     }
                     None => bounds.push(GenericBound::maybe_sized(cx)),
                 }
+                // Move bounds that are (likely) directly attached to the parameters of the
+                // (generic) associated type from the where clause to the respective parameter.
+                // There is no guarantee that this is what the user actually wrote but we have
+                // no way of knowing.
+                let mut where_predicates = Vec::new();
+                for mut pred in generics.where_predicates.drain(..) {
```

Comment:
Wouldn't a `into_iter()` achieve the same result?

Reply:
Ah, you are right! Not sure why I thought I couldn't move out of `generics.where_predicates`. Pushing fix right away…